lms/djangoapps/verify_student/views.py

```diff
@@ -305,7 +317,9 @@ def get(
         # with a paid course mode (such as "verified").
         # For this reason, every paid user is enrolled, but not
         # every enrolled user is paid.
-        already_verified = self._check_already_verified(request.user)
+        # If the course mode is not verified(i.e only paid) then already_verified is always True
+        already_verified = self._check_already_verified(request.user) \
+            if CourseMode.is_verified_mode(course_mode) else True
```

Comment:
already_verified flag is also used in _redirect_if_necessary() function so I think it is okay to check here and pass the flag to both functions